apps/email/js/html_cache.js

```diff
@@ -37,56 +39,66 @@ exports.cloneAsInertNodeAvoidingCustomElementHorrors = function(node) {
  * This method claims all cookie keys that have pattern
  * /htmlc(\d+)/
  */
-exports.save = function htmlCacheSave(html) {
+exports.save = function htmlCacheSave(moduleId, html) {
+  // Only save the last part of the module ID as the cache key. This is specific
+  // to how email lays out all card modules in a 'cards/' module ID prefix, and
+  // with all / and underscores turned to dashes for component names.
+  var id = exports.moduleIdToKey(moduleId);
+
   var langDir = document.querySelector('html').getAttribute('dir');
-  html = encodeURIComponent(HTML_COOKIE_CACHE_VERSION +
+  html = HTML_CACHE_VERSION +
                             (langDir ? ',' + langDir : '') +
-                            ':' + html);
+                            ':' + html;
 
-  // Set to 20 years from now.
-  var expiry = Date.now() + (20 * 365 * 24 * 60 * 60 * 1000);
-  expiry = (new Date(expiry)).toUTCString();
+  // console.log('SAVING: ' + 'html_cache_' + id + ': ' + html);
```

Comment:
nit: this commented out line seems potentially mooted by the active console.log below.